sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1520,6 +1520,11 @@ private Mono<RxDocumentServiceRequest> getCreateDocumentRequest(DocumentClientRe
 
         RxDocumentServiceRequest request = RxDocumentServiceRequest.create(this,
             operationType, ResourceType.Document, path, requestHeaders, options, content);
+
+        if (operationType.isWriteOperation() && options.getNonIdempotentWriteRetriesEnabled()) {
```

Comment:
I wonder how we handle StoredProcedure. as `ExecuteJavaScript` also belong to write operation

Reply:
True - will need ot be suppressed - similar as patch - no chacne to identify whether it would be safe form idempotency perspective or not

Reply:
Like disabled unless opted-in in request options

Reply:
make sense